src/model/clients/HttpClient.js

```diff
@@ -114,13 +114,21 @@ export class HttpClient {
 
     await this.onBeforeRequest(this.requestConfig);
 
-    return axios({
-      baseURL,
-      timeout,
-      headers,
-      url,
-      method,
-      data,
-    });
+    try {
+      await this.onBeforeRequest(this.requestConfig);
+
+      const response = await axios({
+        baseURL,
+        timeout,
+        headers,
+        url,
+        method,
+        data,
+      });
+
+      return response.data;
+    } catch (err) {
+      return err.response.data;
```

Comment:
Please beware that this will return a resolved promise, not a rejected one.

Reply:
And this is what I don't get... AFAIK when a promise return an error it is automatically rejected, right?

Another consideration would be that here we will probably need to return the whole response if we want to chech statuses in thunks, so we will have either to depend a little bit on axios or keep providing abstractions for whatever use cases we want.

Reply:
Check this out:
https://repl.it/@axbarsan/failed-promises
Because we are catching the failed promise, it is no longer rejected.  If we return in the `catch` block, it is the same as returning in the `try` block. The error is already caught, so the promise will resolve.

Reply:
True.

I prefer my first proposal, but if you don't want to use `.data` in actions, which I find reasonable, then we can throw here and catch later:

```javascript
try {
// ...
} catch (err) {
      throw Error(err.response.data);
    }
```

But I think we should be throwing `err.response` so we pass the full error to action creators to do whatever check we want to do there.

Reply:
To be able to use response codes and what else in actions, and not depend on `axios`, can you please create a custom response class, that we can pass around and use?